crates/polars-lazy/src/tests/io.rs

```diff
@@ -440,37 +440,66 @@ fn slice_at_union(lp_arena: &Arena<ALogicalPlan>, lp: Node) -> bool {
 }
 
 #[test]
-fn test_csv_globbing() -> PolarsResult<()> {
+fn test_csv_globbing() {
     let glob = "../../examples/datasets/foods*.csv";
-    let full_df = LazyCsvReader::new(glob).finish()?.collect()?;
+    let full_df = LazyCsvReader::new(glob)
+        .finish()
+        .unwrap()
+        .collect()
+        .unwrap();
 
     // all 5 files * 27 rows
     assert_eq!(full_df.shape(), (135, 4));
-    let cal = full_df.column("calories")?;
-    assert_eq!(cal.get(0)?, AnyValue::Int64(45));
-    assert_eq!(cal.get(53)?, AnyValue::Int64(194));
-
-    let glob = "../../examples/datasets/*.csv";
-    let lf = LazyCsvReader::new(glob).finish()?.slice(0, 100);
+    let cal = full_df.column("calories").unwrap();
+    assert_eq!(cal.get(0).unwrap(), AnyValue::Int64(45));
+    assert_eq!(cal.get(53).unwrap(), AnyValue::Int64(194));
+}
 
-    let df = lf.clone().collect()?;
+#[test]
+fn test_csv_globbing_2() {
+    let lf = LazyCsvReader::new("../../examples/datasets/*.csv")
+        .finish()
+        .unwrap()
+        .slice(0, 100);
+    let df = lf.clone().collect().unwrap();
     assert_eq!(df.shape(), (100, 4));
-    let df = LazyCsvReader::new(glob).finish()?.slice(20, 60).collect()?;
-    assert!(full_df.slice(20, 60).equals(&df));
+}
+
+#[test]
+fn test_csv_globbing_3() {
+    let df_foods = LazyCsvReader::new("../../examples/datasets/foods*.csv")
+        .finish()
+        .unwrap()
+        .collect()
+        .unwrap()
+        .slice(20, 60);
+    let df_all = LazyCsvReader::new("../../examples/datasets/*.csv")
+        .finish()
+        .unwrap()
+        .slice(20, 60)
+        .collect()
+        .unwrap();
+    assert!(df_foods.equals(&df_all));
+}
```

Comment:
This looks like we're testing if slicing before and after collect produces the same result. We could add these tests for all file formats in `test_scan.py` and remove this one.

Reply:
Let's leave it here. Though there is redundancy, in refactors it is nice to have (incomplete) Rust tests to avoid compiling the whole.